File: jvm/src/main/java/ch/neukom/advent2022/day5/Part2.java
package ch.neukom.advent2022.day5;

import java.io.IOException;

import ch.neukom.advent2022.day5.Storage.StorageType;
import ch.neukom.advent2022.util.InputResourceReader;

public class Part2 {
    public static void main(String[] args) throws IOException {
        try (InputResourceReader reader = new InputResourceReader(Part2.class)) {
            run(reader);
        }
    }

    private static void run(InputResourceReader reader) {
        Util.solvePuzzle(reader, StorageType.CrateMover9001);
    }
}

File: jvm/src/main/kotlin/ch/neukom/advent2022/day1/Part2.kt
package ch.neukom.advent2022.day1

import ch.neukom.advent2022.util.InputResourceReader
import java.util.stream.Collectors

object Part2

fun main() {
    InputResourceReader(Part2.javaClass).use(::run)
}

private fun run(reader: InputResourceReader) {
    val input = reader.readInput().collect(Collectors.joining("\n"))
    val sumOfThreeMostCarried = input.splitToSequence("\n\n")
        .map { parseNumbers(it) }
        .map { it.sum() }
        .sortedDescending()
        .take(3)
        .sum()
    println("The sum of the three highest amounts of calories carried is $sumOfThreeMostCarried")
}

private fun parseNumbers(it: String) = it.splitToSequence("\n").map { it.toLong() }
